### ISSUE TITLE
Deployment Error: Insufficient permissions to (re)configure a trigger

### ISSUE THREAD

**Issue by jloosli** [firebase/firebase-functions#419]
### Related issues

#39 

### [REQUIRED] Version info

**node:**
v10.9.0

**firebase-functions:**
2.2.0

**firebase-tools:**
6.4.0

**firebase-admin:**
6.0.0

### [REQUIRED] Test case

Not sure how to do this since it's tied to a specific project

### [REQUIRED] Steps to reproduce

`firebase deploy --only functions`

### [REQUIRED] Expected behavior

Deploy firebase functions without error.

### [REQUIRED] Actual behavior

> % firebase deploy --only functions                                                     
> === Deploying to 'xxxxxx'...
> i  deploying functions
> Running command: npm --prefix "$RESOURCE_DIR" run lint
> functions@ lint /media/xxxxx/storage/projects/xxxxxx/functions
> tslint --project tsconfig.json
> Running command: npm --prefix "$RESOURCE_DIR" run build
> functions@ build /media/xxxxx/storage/projects/xxxxxxx/functions
> tsc
> ✔  functions: Finished running predeploy script.
> i  functions: ensuring necessary APIs are enabled...
> ✔  functions: all necessary APIs are enabled
> i  functions: preparing functions directory for uploading...
> i  functions: packaged functions (122.93 KB) for uploading
> ✔  functions: functions folder uploaded successfully
> i  functions: updating Node.js 8 function registrationUpload(us-central1)...
> ⚠  functions[registrationUpload(us-central1)]: Deployment error.
> Insufficient permissions to (re)configure a trigger (permission denied for bucket registration). Please, give owner permissions to the editor role of the bucket and try again.

> Functions deploy had errors with the following functions:
>        registrationUpload

> To try redeploying those functions, run:
>    firebase deploy --only functions:registrationUpload

> To continue deploying other features (such as database), run:
>    firebase deploy --except functions

> Error: Functions did not deploy properly.

In firebase functions, the only function I have is as follows (using typescript):

```
import * as functions from 'firebase-functions';

const registrationUpload = functions.storage.bucket('registration').object()
  .onFinalize((object, context) => {
    console.log(object);
    console.log(context);
  });

export {registrationUpload};
```

#### What I've tried so far:

* Use runtime of node 6 and 8: No change 
* I followed the instructions for using `gsutil` to change instructions (see https://github.com/firebase/firebase-functions/issues/39#issuecomment-295773400 and https://github.com/firebase/firebase-functions/issues/39#issuecomment-295799385) and I get the same results. 
* When running `gsutil list`, I noticed that the storage paths listed were for a different firebase project. I ran `gcloud config set project xxxxxx` and verified that the current gcloud project was the one I was working on using both `gcloud config configurations list` and `gsutil list`. Still same results.
* Confirmed that I am the owner of the project through IAM
* Confirmed that I'm logged in through `firebase login`

### Were you able to successfully deploy your functions?

No (see above)

**Comment by jloosli** [comment]
OK...I figured it out. I removed the `.bucket('registration')` from the code. I was trying to watch a subfolder called 'registration' in my default bucket. In my mind, the `.bucket()` was used like `.collection()` or `.doc()` in firestore, where it denoted a hierarchy. It looks like I just watch the default bucket, then look at the object.id or object.name to determine if it's in the 'registration' subfolder.
